x/stake/client/rest/query.go

```diff
@@ -75,13 +77,7 @@ func delegationHandlerFn(ctx context.CoreContext, cdc *wire.Codec) http.HandlerF
 			return
 		}
 
-		var delegation stake.Delegation
-		err = cdc.UnmarshalBinary(res, &delegation)
-		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(fmt.Sprintf("couldn't decode delegation. Error: %s", err.Error())))
-			return
-		}
+		delegation := types.UnmarshalDelegation(cdc, key, res)
```

Comment:
Do we want to handle the error here? The remote node could return nonsense.

Reply:
we want error handling on unmarshalling because we making a remote call, where although we expect that the node is running compatible software, there is no guarantee of it. Thus it’s not totally unexpected behaviour that the request could return garbuleygoop and we shouldn’t panic on it … is this more-less your thinking?

K - makes sense to me

Reply:
addressed